static/index.html

```diff
@@ -1,104 +1,106 @@
+    <meta charset="utf-8" />
     <style>
-      * { margin: 0; padding: 0; box-sizing: border-box; }
-      body { font: 13px Helvetica, Arial; }
-      form { background: #000; padding: 3px; position: fixed; bottom: 0; width: 100%; }
-      form input { border: 0; padding: 10px; width: 90%; margin-right: .5%; }
-      form button { width: 9%; background: rgb(130, 224, 255); border: none; padding: 10px; }
-      #messages { list-style-type: none; margin: 0; padding: 0; }
-      #messages li { padding: 5px 10px; }
-      #messages li:nth-child(odd) { background: #dedede; }
-      #messages li:last-child { background: #aea; }
+      * {
+        box-sizing: border-box;
+        margin: 0;
+        padding: 0;
+      }
+      *:focus {
+        outline: none;
+      }
+      html {
+        font-family: Helvetica, Arial;
+        margin: 0 auto;
+        max-width: 32rem;
+      }
+      body {
+        align-items: center;
+        display: flex;
+        flex-direction: column;
+        height: 100vh;
+        justify-content: center;
+        padding: 4rem 1rem;
+      }
+      h1 {
+        font-weight: 400;
+        letter-spacing: 0.5rem;
+      }
       section {
-        background-color: #eee;
-        border: 3px dashed #888; border-radius: 10px;
-        margin: 30px; margin-bottom: 80px;
-        padding: 5px;
+        flex-grow: 1;
+        margin: 2rem 0;
+        overflow: auto;
+        width: 100%;
+      }
+      section p {
+        padding: 0.5rem 0;
+      }
+      form {
+        display: flex;
+        width: 100%;
+      }
+      form input {
+        border: 1px solid #ccc;
+        flex-grow: 1;
+        padding-left: 0.5rem;
+      }
+      form button {
+        background: black;
+        border: none;
+        color: white;
+        padding: 0.5rem 1rem;
       }
     </style>
   </head>
   <body>
+    <h1>Foxtrot</h1>
+
+    <section id="messages"></section>
 
-    <!-- [START gae_flex_websockets_form] -->
-    <h1>Websockets Chat Demo</h1>
-    
-    <form id="chat-form">
-      <input type="text" id="chat-text" autocomplete="off" placeholder="Enter some text...">
+    <form id="form">
+      <input id="message-input" autocomplete="off" />
       <button type="submit">Send</button>
     </form>
 
-    <section>
-
-      <ul id="messages"></ul>
-    </section>
-
-    <!-- [END gae_flex_websockets_form] -->
-
-    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
     <script>
-    // [START gae_flex_websockets_js]
-    $(function() {
-      /* If the main page is served via https, the WebSocket must be served via
-         "wss" (WebSocket Secure) */
-      var scheme = window.location.protocol == "https:" ? 'wss://' : 'ws://';
-      var webSocketUri =  scheme
-                          + window.location.hostname
-                          + (location.port ? ':'+location.port: '')
-                          + '/ws';
+      ;(function () {
```

Comment:
what weird syntax is this `;(...)`. sigh, javascript. no-one loves you.

Reply:
yeah, not really 100% sure actually why this is here.
I've taken it from https://github.com/nhooyr/websocket/blob/master/examples/chat/index.js when I removed jquery.
I think it is to do with other libraries coming before potentially being dodgy.
everything's wrapped in a function so we don't pollute the global namespace.

Reply:
Here are some thoughts on the subject:
https://stackoverflow.com/questions/8228281/what-is-the-function-construct-in-javascript

Reply:
ok. i'm not going to dicker around over javascript. The only place I saw in that SO question where `;(...)` was mentioned was in a comment that said "aaand the winner is `;(function(){}());`". Maybe I was meant to read and understand the question and answers in depth and then I would understand why the semicolon by induction, but i'm not going to. so ok.